### ISSUE TITLE
Add pattern synonyms for Data.Sequence

### ISSUE THREAD

**Issue by treeowl** [haskell/containers#186]
Now that GHC's support for pattern synonyms is reaching a usable level, I think it would be great to offer (bidirectional) pattern synonyms for manipulating the ends of sequences. These would let users avoid the pain of explicitly applying `viewl` or `viewr`. I suggest the names `:>>` and `:<<`, but I'm not particularly tied to them. Something like this:

```haskell
pattern x :<< xs <- viewl -> x :< xs where
  x :<< xs = x <| xs
pattern xs :>> xs <- viewr -> xs :> x where
  xs :>> x = xs |> x
pattern EmptyS <- null -> True where
  EmptyS = empty
```

Then a user could just write

```haskell
f :: Seq Int -> Seq Int
f EmptyS = EmptyS
f (xs :>> x) = xs :>> (x + 1)
```

**Comment by treeowl** [comment]
The names `:|>` and `:<|` would make more sense, but they take longer to type.

**Comment by foxik** [comment]
I always wanted something like that :-)

I vote for the "known operator prefixed with colon", i.e., `:|>` and `:<|`.

**Comment by treeowl** [comment]
I'd also want to use `Empty` instead of `EmptyS`. The easiest way would be to rename the `Empty` constructor, but moving the implementation to an `Internal` module would certainly enable that too.

**Comment by foxik** [comment]
I think we can rename the `Empty` constructor to `EmptyS` and use `Empty` for the pattern.

**Comment by treeowl** [comment]
This is implemented in #194.
